src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+        /// <param name="entityFields">All the fields belonging to the entity</param>
+        /// <returns>Primary Key fields</returns>
+        private static Dictionary<string, object?> FetchPrimaryKeyFields(ISqlMetadataProvider sqlMetadataProvider, string entityName, Dictionary<string, object?> entityFields)
+        {
+            Dictionary<string, object?> pkFields = new();
+            SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(entityName);
+            foreach (string primaryKey in sourceDefinition.PrimaryKey)
+            {
+                if (sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? name))
+                {
+                    if (entityFields.TryGetValue(name, out object? value))
+                    {
+                        pkFields.Add(primaryKey, value);
+                    }
+                }
+            }
+
+            return pkFields;
+        }
+
+        /// <summary>
+        /// Helper method to populate the linking table referencing fields.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="fkDefinition">Foreign Key metadata constructed during engine start-up</param>
+        /// <param name="multipleCreateStructure">Wrapper object assisting with the multiple create operation.</param>
+        /// <param name="computedRelationshipFields">Relationship fields obtained as a result of creation of current or higher level entity item.</param>
+        /// <param name="isLinkingTable">Indicates whether referencing fields are populated for a linking entity.</param>
+        /// <param name="entityName">Name of the entity.</param>
+        private static void PopulateReferencingFields(ISqlMetadataProvider sqlMetadataProvider, MultipleCreateStructure multipleCreateStructure, ForeignKeyDefinition fkDefinition, Dictionary<string, object?>? computedRelationshipFields, bool isLinkingTable, string? entityName = null)
+        {
+            if (computedRelationshipFields is null)
+            {
+                return;
+            }
+
+            for (int i = 0; i < fkDefinition.ReferencingColumns.Count; i++)
+            {
+                string referencingColumnName = fkDefinition.ReferencingColumns[i];
+                string referencedColumnName = fkDefinition.ReferencedColumns[i];
+                string exposedReferencedColumnName;
+                if (isLinkingTable)
+                {
+                    multipleCreateStructure.LinkingTableParams![referencingColumnName] = computedRelationshipFields[referencedColumnName];
+                }
+                else
+                {
+                    if (entityName is not null
+                        && sqlMetadataProvider.TryGetExposedColumnName(entityName, referencedColumnName, out string? exposedColumnName)
+                        && !string.IsNullOrWhiteSpace(exposedColumnName))
```

Comment:
Can `exposedColumnName` be null/whitespace?

Reply:
No, it cannot be. Removed that check